toter.js

```diff
@@ -141,145 +155,166 @@ switch(command) {
+            type: 'marketplace',
+            source: '/cmp/api/storage/buckets/' + config[selectedRegion].widget_json.id + '/' + entry
+        })
+
+        let widgetID = curlObj.id
+        delete curlObj.id
+        let curlData = curlHelper('/api/apps/widgets/' + config[selectedRegion].widget_json.id, curlObj, 'put')
+        config[selectedRegion].widget_json = Object.assign({id: widgetID}, JSON.parse(curlData))
+        stripFields(config[selectedRegion].widget_json)
+    }
+
+    if(!config[selectedRegion].app_json || config[selectedRegion].app_json && !config[selectedRegion].app_json.id) {
+        console.log('Please run setup first - app has no ID')
+    } else {
+        config[selectedRegion].app_json = JSON.parse(curlHelper('/api/apps', config[selectedRegion].app_json))
+
+        stripFields(config[selectedRegion].app_json)
+    }
+    fs.writeFileSync('config.json', JSON.stringify(config, null, 4))
 }
+
+function upload() {
+    if(config[selectedRegion].widget_json && config[selectedRegion].widget_json.id) {
+        // Needs to be two commands due to the first and second curls
+        // having different data bodies
+        curlHelper('/api/storage/buckets/' + config[selectedRegion].widget_json.id, {
+            type: 'public'
+        }, 'put')
+        curlHelper('/api/storage/archive/' + config[selectedRegion].widget_json.id, false, 'put', 'x-tar')
+    } else {
+        console.log('Run setup first')
+    }
+}
+
+function submit() {
+    if(config[selectedRegion].app_json && config[selectedRegion].app_json.id) {
+        curlHelper('/api/apps/' + config[selectedRegion].app_json.id + '/submit', false, 'post')
+    } else {
+        console.log('You don\'t appear to have an app id')
+    }
+}
+
+function approve() {
+    if(config[selectedRegion].app_json && config[selectedRegion].app_json.id) {
+        curlHelper('/api/apps/' + config[selectedRegion].app_json.id + '/approve', false,
+            'post')
+    } else {
+        console.log('You don\'t appear to have an app id')
+    }
+}
+
+function help() {
+    console.log([
+        'usage: toter [command] [options]',
+        '',
+        'commands:',
+        '  config            Configure toter with your auth credentials',
+        '    -u --url        Set the region url',
+        '    -k --key        Set the key',
+        '    -s --secret     Set the secret',
+        '    -n --new        Set the region name - used to specify different regions',
+        '  setup             Setup your repository as a Marketplace widget',
+        '  upload            Upload the contents of your dist/ folder to StormDrive',
+        '  submit            Submit your widget for review',
+        '  help              Print this list and exit',
+        '',
+        'options:',
+        '  -r --region      Region to use other than default',
+        ''
+    ].join('\n'))
+    process.exit()
+}
+
+const commands = {
+    config: setConfig,
```

Comment:
Was there a reason to call this setConfig, different to all the other methods?

Reply:
Yeah, `config` is already defined as it is the config object. I had to call one of them something different.